client/my-sites/plans/jetpack-plans/product-store/types.ts

```diff
@@ -1,23 +1,27 @@
-import { BasePageProps, Duration } from '../types';
+import { QueryArgs, Duration } from '../types';
+
+export type ViewType = 'products' | 'bundles';
 
 export interface ProductStoreBaseProps {
 	siteId: number | null;
 }
 
-export interface ProductStoreProps extends Pick< BasePageProps, 'urlQueryArgs' > {
+export interface ProductStoreQueryArgs extends QueryArgs {
+	view: ViewType;
+}
```

Comment:
This should fix your TS error.

```suggestion
export type ProductStoreQueryArgs = QueryArgs & {
	view?: ViewType;
};
```